client/client-multi/client-c/src/kaa/platform-impl/common/common_key_utils.c

```diff
@@ -0,0 +1,391 @@
+kaa_error_t kaa_init_keys(void)
+{
+    /* Initialization should be performed only once */
+    static bool initialized = false;
+    if (!initialized) {
+        /* Generate session key */
+        if (init_aes_key(keys.session_key, KAA_SESSION_KEY_LENGTH)) {
+            return KAA_ERR_BADDATA;
+        }
+
+        keys.session_key_length = KAA_SESSION_KEY_LENGTH;
+        if (mbedtls_pk_parse_key(&pk_context_, kaa_rsa_private_key,
+                                 KAA_RSA_PRIVATE_KEY_LENGTH, NULL, 0)) {
+            return KAA_ERR_BADDATA;
+        }
+        initialized = true;
+    }
+
+    return KAA_ERR_NONE;
+}
+
+void kaa_deinit_keys(void)
+{
+    mbedtls_pk_free(&pk_context_);
+}
+
+void ext_get_endpoint_session_key(uint8_t **buffer, size_t *buffer_size)
+{
+    if (!buffer || !buffer_size) {
+      return;
+    }
+    *buffer = keys.session_key;
+    *buffer_size = keys.session_key_length;
+}
+
+/* Get encrypted session key (enctypted with remote key) */
+kaa_error_t ext_get_encrypted_session_key(uint8_t **buffer, size_t *buffer_size,
+                                          uint8_t *remote_key, size_t remote_key_size)
+{
+    if (!buffer || !buffer_size || !remote_key || !remote_key_size) {
+      return KAA_ERR_BADPARAM;
+    }
+
+    mbedtls_pk_context local_ctx;
+    mbedtls_pk_init(&local_ctx);
+    int err = set_rsa_public_key(&local_ctx, remote_key, remote_key_size);
+    if (err) {
+      goto exit;
+    }
+    err = rsa_encrypt(&local_ctx, keys.session_key,
+                      KAA_SESSION_KEY_LENGTH, keys.encrypted_session_key);
+    if (err) {
+      goto exit;
+    }
+
+    keys.encrypted_session_key_length = ENC_SESSION_KEY_LENGTH;
+
+    *buffer = keys.encrypted_session_key;
+    *buffer_size = keys.encrypted_session_key_length;
+
+exit:
+    mbedtls_pk_free(&local_ctx);
+    return err ? KAA_ERR_BADDATA : KAA_ERR_NONE;
+}
+
+__attribute__((weak))
+void ext_get_endpoint_public_key(uint8_t **buffer, size_t *buffer_size)
+{
+    if (!buffer || !buffer_size) {
+      return;
+    }
+
+    *buffer = kaa_rsa_public_key;
+    *buffer_size = KAA_RSA_PUBLIC_KEY_LENGTH;
+}
+
+void ext_get_encrypted_data_size(size_t input_size, size_t *output_size)
+{
+    if (!input_size || !output_size) {
+        *output_size = 0;
+        return;
+    }
+    *output_size = !(input_size % AES_ECB_ENCRYPTION_CHUNCK_SIZE) ?
+                   input_size + AES_ECB_ENCRYPTION_CHUNCK_SIZE :
+                   (AES_ECB_ENCRYPTION_CHUNCK_SIZE -
+                    (input_size % AES_ECB_ENCRYPTION_CHUNCK_SIZE) + input_size);
+}
+
+kaa_error_t ext_encrypt_data(uint8_t *input, size_t payload_size, uint8_t *output)
+{
+    if (!input || !payload_size || !output) {
+      return KAA_ERR_BADPARAM;
+    }
+
+    /* Adding PKCS7 padding */
+    size_t enc_data_size;
+    ext_get_encrypted_data_size(payload_size, &enc_data_size);
+    uint8_t padding = enc_data_size - payload_size;
+    memset(input + payload_size, padding, padding);
```

Comment:
Please leave here a comment, explaining why padding is `uint8_t` and why pad data is equal to pad size